examples/src/bin/32khz_xtal.rs

```diff
@@ -0,0 +1,50 @@
+//! This turns on the 32 khz xtal and then just prints hello world to UART. Requires a special bootloader
```

Comment:
I don't see the value in this test. For one, what's "a special bootloader" that is required for it? Next, what does this example show, that changing the clock source doesn't break the whole system?

Reply:
The bootloader needs to be build with `CONFIG_RTC_CLK_SRC_EXT_CRYS=y`

Well yes, also allows to measure the sine wave with an osciloscope

Reply:
Can we select the clock source here, instead of relying on the bootloader?

Reply:
Not without losing some optimization https://github.com/espressif/esp-idf/blob/59e18382702b6986be3d3f55e9ac7763c1397cf7/components/bootloader_support/src/bootloader_clock_init.c#L82-L90

And maybe something more is behind it, I don't know

Reply:
`if (!rtc_clk_32k_enabled()) {`

Unless that takes long to decide, this is safe to do. If the bootloader enabled the clock source, and the user selects in in esp-hap, we can skip initialization. Otherwise, in the future we might allow [other bootloaders](https://github.com/esp-rs/esp-hal/issues/1973) which may or may not set up the clock source, so I'd say let's enable the clock ourselves, if it's not done already.

Reply:
Please note it's inside the macro statement `#if CONFIG_ESP_SYSTEM_RTC_EXT_XTAL`

Well maybe it is safe but we lose some optimization, I don't think we should lose any performance because at default we don't control the bootloader fully. In the future we probably will by using a rust based bootloader so it will work the same just like now, but without the hassle of recompiling it

Reply:
> I don't think we should lose any performance because at default we don't control the bootloader fully

Let's prefer the option that **works** by default. If the user has problems with the startup time, they can use a custom bootloader.

Reply:
Making it work only in the application would require changing the code for it to wait to start up, not only is it not efficient but also I don't know how to do it, I looked at a lot of esp-idf code and haven't really found anything that waits for it, I ques booting the bootloader is enough time for it to start after all. Using a fixed time is also bad, would require cpu dependent testing.

This is simply not the way to do it...

Reply:
Fine. At least please print a warning or panic if the bootloader forgot to enable the clock source.

Reply:
Ok, I will try to find a way to detect it (probably just failing the calibration step)